File: armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Exercism.CSharp.Solutions.ArmstrongNumbersExercise
{
    public static class ArmstrongNumbers
    {
        public static bool IsArmstrongNumber(int number) =>
            number == number
                        .ToString()
                        .Select(c => c - '0')
                        .Sum(n => Math.Pow(n, number.ToString().Length));
    }
}

File: c/armstrong-numbers/src/armstrong_numbers.c
#include <math.h>

#include "armstrong_numbers.h"

static int count_digits(int number);

bool is_armstrong_number(int candidate)
{
    int result = 0;
    int expected = candidate;
    int digits = count_digits(candidate);
    int factor = digits;

    while (digits--)
    {
        result += pow(candidate % 10 , factor);
        candidate /= 10;
    }
    
    return result == expected;
}

static int count_digits(int number)
{
    int digits = 1;
    while (number /= 10) { digits++; }
    return digits;
}